docarray/array/array_stacked.py

```diff
@@ -168,6 +167,10 @@ def _create_columns(
                 elif issubclass(type_, AbstractTensor):
                     columns[field_to_stack] = type_.__docarray_stack__(to_stack)  # type: ignore # noqa: E501
 
+        for field_name, column in columns.items():
+            for doc, val in zip(docs, column):
```

Comment:
This is doing a second iteration over all documents, I find that a bit concerning from a performance perspective. Is there a way to eagerly assign ever docs view on the fly, in the loop above?

Reply:
> One thing though that is stupid in my implementation is that I should switch the two for loop line 170, 171

Actually I think that this makes zero difference

Reply:
Okay I think there is a way to do only one loop.

We could:
* pre-allocate the data for the tensor column (we can predict the shape and the type by looking at the first row of the da)
* when collecting the actual data we could replace the data with a view to the pre alocated column 
* when stacking put the data into the preallocate column in place. Therefore each document has already a view to the good part of the column and we don't need iterate again

Reply:
what do you mean by pre-allocate?

Reply:
Here is the hacky idea that I had:

While doing the first loop over all the documents, we already know the index that every view of every document will have: it is the same index as the document has in the documentarray.
Thus we can modify its getattr directly there to index into the big tensor in the right place. Of course at this moment in time the big tensor doesnt exist yet, but that's not an issue, it will be created later.

what i dont like about this is the messing with getattr, but there is probably a cleaner way to follow the same general idea.
I think your idea is also along those lines?

Reply:
What you are saying is basically what we did before

My idea is just to pre-alocate so that we know in advance where the column data will stay so that we can get the view (pointer kind of ) to the right place even before the actual data is created

Reply:
tbh I think it is okay not to do the optimization now

Reply:
I think we are basically talking about the same idea ^^ but let's leave it for now, as premature optimization is the root of all evil, as we all know ;)